sky/core.py

```diff
@@ -800,13 +800,17 @@ def spot_queue(refresh: bool,
     stop_msg = ''
     if not refresh:
         stop_msg = 'To view the latest job table: sky spot queue --refresh'
-    controller_status, handle = backend_utils.is_controller_up(
-        controller_type=controller_utils.Controllers.SPOT_CONTROLLER,
-        stopped_message=stop_msg)
+    try:
+        handle = backend_utils.is_controller_up(
+            controller_type=controller_utils.Controllers.SPOT_CONTROLLER,
+            stopped_message=stop_msg)
+    except exceptions.ClusterNotUpError as e:
+        if not refresh:
+            raise
+        handle = None
+        controller_status = e.cluster_status
 
-    if (refresh and controller_status in [
-            status_lib.ClusterStatus.STOPPED, status_lib.ClusterStatus.INIT
-    ]):
+    if refresh and handle is not None:
```

Comment:
Can we double check all these cases, maybe with some smoke tests (if these cases are important enough to prevent regressions)? 

For example, I have a STOPPED spot controller. In this PR,
```
 » spst -r
Fetching managed spot job statuses...
Managed spot jobs
No in progress jobs.
```
On master,
```
» spst -r
Fetching managed spot job statuses...
⠋  Checking spot jobs - restarting controller
```